### ISSUE TITLE
NPE in queryRenderedFeatures

### ISSUE THREAD

**Issue by rm3dbeckers** [mapbox/mapbox-gl-native#11547]
**Platform:** Android
**Mapbox SDK version:** 5.5.1

### Steps to trigger behavior

 1. ```queryRenderedFeatures``` in ```onMapClick```

### Expected behavior

Get an empty ```List<Feature>```.
(Of course generally I do use this with actual layers and would expect a non-empty list, based on the click position, but I always do get the NPE)

### Actual behavior

A NullPointerException is thrown.

### Additional information

I am currently becoming desperate over this. I got it working before, but with a recent code change (I ported my code from java to kotlin) I no longer can get this to work and I am not able to find out what I changed.

Even an extremely simple test program throws this NPE for me:

Note that I also tried this exact test with java instead of kotlin and the result is the same:
The line mMapboxMap.queryRenderedFeatures(rectF, null as Filter.Statement?, null as String?) throws an NPE:

```
03-28 12:53:25.681 24740 24740 D AndroidRuntime: Shutting down VM
03-28 12:53:25.682 24740 24740 E AndroidRuntime: FATAL EXCEPTION: main
03-28 12:53:25.682 24740 24740 E AndroidRuntime: Process: com.test.mapboxglquerytest, PID: 24740
03-28 12:53:25.682 24740 24740 E AndroidRuntime: java.lang.NullPointerException
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.mapbox.mapboxsdk.maps.NativeMapView.nativeQueryRenderedFeaturesForBox(Native Method)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.mapbox.mapboxsdk.maps.NativeMapView.queryRenderedFeatures(NativeMapView.java:819)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.mapbox.mapboxsdk.maps.MapboxMap.queryRenderedFeatures(MapboxMap.java:2232)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.test.mapboxglquerytest.MainActivity.onLocationClick(MainActivity.kt:41)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.test.mapboxglquerytest.MainActivity.access$onLocationClick(MainActivity.kt:13)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.test.mapboxglquerytest.MainActivity$onCreate$1$1.onMapClick(MainActivity.kt:29)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.mapbox.mapboxsdk.maps.MapGestureDetector.notifyOnMapClickListeners(MapGestureDetector.java:483)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.mapbox.mapboxsdk.maps.MapGestureDetector$GestureListener.onSingleTapConfirmed(MapGestureDetector.java:380)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at android.view.GestureDetector$GestureHandler.handleMessage(GestureDetector.java:312)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:102)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:154)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6692)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
03-28 12:53:25.682 24740 24740 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
```

This is the MainActivity.kt:
```
package com.test.mapboxglquerytest

import android.graphics.RectF
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.mapbox.mapboxsdk.Mapbox
import com.mapbox.mapboxsdk.geometry.LatLng
import com.mapbox.mapboxsdk.maps.MapboxMap
import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
import com.mapbox.mapboxsdk.style.layers.Filter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private lateinit var mMapboxMap: MapboxMap

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        Mapbox.getInstance(this, getResources().getString(R.string.mapbox_access_token))

        setContentView(R.layout.activity_main)

        mapView.onCreate(savedInstanceState)

        mapView.getMapAsync(OnMapReadyCallback {
            mMapboxMap = it
            it.addOnMapClickListener {
                onLocationClick(it)
            }
        })
    }

    private fun onLocationClick(latLng: LatLng) {
        val pointf = mMapboxMap.projection.toScreenLocation(latLng)

        var searchRadius = 15f
        val rectF = RectF(pointf.x - searchRadius, pointf.y - searchRadius, pointf.x + searchRadius, pointf.y + searchRadius)

        val featureList = mMapboxMap.queryRenderedFeatures(rectF, null as Filter.Statement?, null as String?)

    }

    override fun onStart() {
        super.onStart()
        mapView.onStart()
    }

    override fun onStop() {
        super.onStop()
        mapView.onStop()
    }

    override fun onResume() {
        super.onResume()
        mapView.onResume()
    }

    override fun onPause() {
        super.onPause()
        mapView.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()
        mapView.onDestroy()
    }

    override fun onLowMemory() {
        super.onLowMemory()
        mapView.onLowMemory()
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        mapView.onSaveInstanceState(outState)
    }
}
```

Just in case it might contain any useful information for you I've also attached the layout file:

```
<?xml version="1.0" encoding="utf-8"?>
<android.support.constraint.ConstraintLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context="com.test.mapboxglquerytest.MainActivity">

    <com.mapbox.mapboxsdk.maps.MapView
        android:id="@+id/mapView"
        android:layout_width="match_parent"
        android:layout_height="match_parent">

    </com.mapbox.mapboxsdk.maps.MapView>

</android.support.constraint.ConstraintLayout>
```

and the build.gradle:

```
apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-android-extensions'

android {
    compileSdkVersion 26

    defaultConfig {
        applicationId "com.test.mapboxglquerytest"
        minSdkVersion 15
        targetSdkVersion 26
        versionCode 1
        versionName "1.0"
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }

}

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jre7:$kotlin_version"
    implementation 'com.android.support:appcompat-v7:26.+'
    implementation 'com.android.support.constraint:constraint-layout:1.0.2'

    api 'com.mapbox.mapboxsdk:mapbox-android-sdk:5.5.1'
}
```

**Comment by tobrun** [comment]
Capturing from @LukasPaczos that he was able to reproduce this with the binary but not from the release agua or either of the 5.5.0/6.0.0-beta.5 binaries. The changes introduced between previous release and 5.5.1 do not touch code. Will look more into this issue.

**Comment by tobrun** [comment]
I'm not able to reproduce with java code using the 5.5.1 binary. Though using the same signature shows the following warning:

<img width="882" alt="screen shot 2018-03-28 at 15 11 06" src="https://user-images.githubusercontent.com/2151639/38050829-53526b4c-329a-11e8-9583-f01135e0f05c.png">

@rm3dbeckers can you test with using the following instead?

```
mMapboxMap.queryRenderedFeatures(rectF)
```

**Comment by rm3dbeckers** [comment]
Oh boy, I got it.
Your remark got me on the right track!
Calling only the RectF call did work without problems and then I did some further tests and actually realized that queryRenderedFeatures(rectF, null as Filter.Statement?, null as String?) does not call the (RectF, Statement, String...) method with (rectF, null, null) but actually with (rectF, null, String[1] {null}) -.-
Obviously I am not too familiar with varargs...
Sorry to bother you and thanks for pointing me into the right direction!
